src/VisualStudio/Core/Def/Implementation/AbstractEditorFactory.cs

```diff
@@ -127,9 +128,24 @@ int IVsEditorFactory.Close()
                 case "Form":
 
                     // We must create the WinForms designer here
-                    const string LoaderName = "Microsoft.VisualStudio.Design.Serialization.CodeDom.VSCodeDomDesignerLoader";
+                    var loaderName = "Microsoft.VisualStudio.Design.Serialization.CodeDom.VSCodeDomDesignerLoader";
+
+                    // If this is a netcoreapp3.0 (or newer), we must create the newer WinForms designer.
+                    // TODO: This check will eventually move into the WinForms designer itself.
+                    const string NetCoreAppPreamble = ".NETCoreApp,Version=v";
+                    if (vsHierarchy.TryGetTargetFrameworkMoniker((uint)VSConstants.VSITEMID.Root, out var targetFrameworkMoniker) &&
```

Comment:
Thanks! Yes, this is definitely a temporary dependency. I did test a multi-targeted project and it appears that IVsHierarchy TFM just returns the first TFM in the `<TargetFrameworks>`. So, it will specify the new loader name if `netcoreapp3.0`, and it'll continue to use the loader name for the Framework WinForms Designer if it's some other TFM. Ultimately, we need a better UX design here, but I think this is good enough for now.

Reply:
An easier way to do this is just throw this in a FrameworkName.

Reply:
Good call. I'll do that

Reply:
Thanks for the suggestion @davkean. Updated to use `FrameworkName`.